lib/active_merchant/billing/gateways/orbital.rb

```diff
@@ -478,6 +499,32 @@ def add_customer_address(xml, options)
         end
       end
 
+      # Payment can be done through either Credit Card or Electronic Check
+      def add_payment_source(xml, payment_source, options = {})
+        if payment_source.instance_of?(ActiveMerchant::Billing::Check)
+          add_echeck(xml, payment_source, options)
+        else
+          add_creditcard(xml, payment_source, options[:currency])
```

Comment:
If we do this in the current code: 
`
        if payment_source.instance_of?(ActiveMerchant::Billing::Check)
          add_echeck(xml, payment_source, options)
        elsif payment_source.instance_of?(ActiveMerchant::Billing::CreditCard) or payment_source.instance_of?(ActiveMerchant::Billing::NetworkTokenizationCreditCard)
          add_creditcard(xml, payment_source, options[:currency])
        end
`

then in that case, Orbital's previously existing two unit test cases are failed as in the previous implementation CurrencyCode needs to be added even when no payment source is passed (please refer to image below):

![image](https://user-images.githubusercontent.com/73054822/103910320-83fdfd80-5126-11eb-99dc-3bf1643968eb.png)

Now, we can have three possibilities here:
1. Keep it as it is so that everything works fine as in the case of no credit card, some required attributes will be added
2. Make changes in previous Orbital code and debug so that it should work even when no payment source is added
3. Make changes in the tests of the previous orbital's implementation

Please suggest on this?

Reply:
If there's existing code that this check would break than let's avoid adding that in. I don't want to mess with any legacy code that could potentially cause a lot more issues to existing customers. Thanks!